lib/src/command/login.dart

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import '../command.dart';
+import '../io.dart' as io;
+
+import '../tokens.dart';
+
+/// Handles the `login` pub command.
+class LoginCommand extends PubCommand {
+  @override
+  String get name => 'login';
+  @override
+  String get description => 'Log in to hosted server.';
```

Comment:
```suggestion
  String get description => 'Log in to third-party pub server.';
```

Let's refer to these as "third-party pub server".